python/ray/serve/http_proxy.py

```diff
@@ -79,12 +79,13 @@ async def _send_request_to_handle(handle, scope, receive, send):
                          f"{MAX_REPLICA_FAILURE_RETRIES} retries.")
         await Response(
             error_message, status_code=500).send(scope, receive, send)
-        return
+        return 500
 
     if isinstance(result, starlette.responses.Response):
         await result(scope, receive, send)
     else:
         await Response(result).send(scope, receive, send)
+    return 200
```

Comment:
Shouldn this be the `status_code` of the Starlette Response object? https://www.starlette.io/responses/ A Serve deployment can return a Starlette Response object and the user code might specify a non-200 status code.

Reply:
actually you can just `return None` here. The return isn't used. The response was sent via the `await Response block`

Reply:
Ah but we're using the return value internally in this PR I think

Reply:
@Hankpipi Why do we always return 200 here?  I agree it should be 200 in the `else` case, because the user hasn't specified a statust code, but in the `if` case, the user has contructed and returned a `starlette.responses.Response` object, and might have specified a status code other than `200`. Shouldn't we extract the status code from the `starlette.responses.Response` object and return that instead of `200`?

Reply:
@architkulkarni, I have changed return value to `None` here which means there's no error in response.

Reply:
Ah that's right, sorry for my carelessness!